zeebe/dynamic-node-id-provider/src/main/java/io/camunda/zeebe/dynamic/nodeid/NodeIdProvider.java

```diff
@@ -169,11 +170,21 @@ private StoredLease.Initialized tryAcquire(final StoredLease lease) {
 
   @Override
   public void close() throws Exception {
-    // release is already submitted to the executor, we can shut it down gracefully.
-    executor.shutdown();
-    // shutdown is taking too much time, let's shut it down by interrupting running tasks.
-    if (!executor.awaitTermination(5, TimeUnit.SECONDS)) {
-      executor.shutdownNow();
+    // use the executor status to avoid closing multiple times
+    if (!executor.isShutdown()) {
+      try {
```

Comment:
:x: Shouldn't we cancel the renewal timer before releasing the lease? Otherwise, renewing and releasing can execute concurrently.